hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/scm/GetScmRatisRolesSubcommand.java

```diff
@@ -38,11 +45,45 @@ public class GetScmRatisRolesSubcommand extends ScmSubcommand {
   @CommandLine.ParentCommand
   private ScmAdmin parent;
 
+  @CommandLine.Option(names = { "--json" },
+      defaultValue = "false",
+      description = "Format output as JSON")
+  private boolean json;
+
   @Override
   protected void execute(ScmClient scmClient) throws IOException {
-    List<String> roles = scmClient.getScmRatisRoles();
-    for (String role: roles) {
-      System.out.println(role);
+    List<String> ratisRoles = scmClient.getScmRatisRoles();
+    if (json) {
+      Map<String, Map<String, String>> scmRoles = parseScmRoles(ratisRoles);
+      System.out.print(
+          JsonUtils.toJsonStringWithDefaultPrettyPrinter(scmRoles));
+    } else {
+      for (String role: ratisRoles) {
+        System.out.println(role);
+      }
+    }
+  }
+
+  private Map<String, Map<String, String>> parseScmRoles(
+      List<String> ratisRoles) {
+    Map<String, Map<String, String>> allRoles = new HashMap<>();
+    for (String role : ratisRoles) {
+      Map<String, String> roleDetails = new HashMap<>();
+      String[] roles = role.split(":");
+      if (roles.length < 2) {
+        err.println("Invalid response received for ScmRatisRoles.");
+        return Collections.emptyMap();
+      }
+      // In case, there is no ratis, there is no ratis role.
+      // This will just print the hostname with ratis port as the address
+      roleDetails.put("address", roles[0].concat(":").concat(roles[1]));
+      if (roles.length == 5) {
```

Comment:
although this would work for now, I feel a better way to solve this is by creating an Object that would capture these variables and then print it right away as  a json.
Adding a new API like List<SCMRole> getScmRatisRoles() and directly print the object. I see OM is using a similar  way using OMRoleInfo.

Reply:
I guess we could also solve this without modifying the current API a lot, Only on  the server side introduce  this API that returns an Object , json Stringify the  object and return it to this current API.

Reply:
I think instead of creating new Map by parsing getScmRatisRoles() , we can use existing API and then use JsonUtils to print it. Something like -

```
List<SCMNodeInfo> scmNodeInfos = SCMNodeInfo.buildNodeInfo(new OzoneConfiguration());
System.out.print(JsonUtils.toJsonStringWithDefaultPrettyPrinter(scmNodeInfos));

```

Reply:
Creating a change on the server side either by stringifying a json object and storing it in peerRoles, or returning an object itself breaks the usage of this method in other places and many tests as well. Currently, everywhere that peerRoles are handled, it has been done by string operations itself. We can defer from making this change right now.

Reply:
> I think instead of creating new Map by parsing getScmRatisRoles() , we can use existing API and then use JsonUtils to print it. 

`SCMNodeInfo.buildNodeInfo(new OzoneConfiguration())` constructs the result based on client config and  does not have the correct ratis role details, There is no API like this that queries the SCM and returns such an Object currently. The existing API returns a String and if we need to change it/add another API  , it would require a change in the proto structure too, I think we could solve it in another jira  by refactoring the current API if really needed.